src/modules/app/app.gateway.ts

```diff
@@ -5,92 +5,134 @@ import {
-    privateRooms: Set<string> = new Set()
-    messages: Map<string, Array<string | Message>> = new Map([[Rooms.All, []]]) //  roomname => [...Message]
-    activeUsers: Set<string> = new Set()
 
     constructor(private readonly appsService: AppService) {}
 
     @SubscribeMessage('getUsersList')
-    getUserList(client: Socket) {
-        client.emit('sendUsersList', [...this.users.entries()])
-        client.emit('activeUsers', [...this.activeUsers.keys()])
+    async getUserList(client: Socket) {
+        const [users, activeUsers] = await Promise.all([
+            this.appsService.getUsersList(),
+            this.appsService.getActiveUsers()
+        ])
+        const userNames = users
+            .map(({ socketId, userName, avatar }) =>
+                [socketId, [ userName, avatar.toString()]])
+        const activeUserNames = activeUsers.map(({ socketId }) => socketId)
+
+        client.emit('sendUsersList', userNames)
+        client.emit('activeUsers', activeUserNames)
     }
 
     @SubscribeMessage('getRoomList')
-    getRoomList(): WsResponse {
-        const roomList = [...this.rooms.keys()].filter(roomName => !this.privateRooms.has(roomName))
+    async getRoomList(): Promise<WsResponse> {
+        const rooms = await this.appsService.getPublicRooms()
+        const roomNames = rooms.map(({ roomName }) => roomName)
 
-        return { event: 'sendRoomList', data: roomList }
+        return { event: 'sendRoomList', data: roomNames }
     }
 
     @SubscribeMessage('newUser')
-    newUser(client: Socket, [userName, avatar]: Array<string>) {
-        const allUsers = [...this.rooms.get(Rooms.All), userName]
-
-        this.users.set(client.id, [userName, avatar])
-        this.rooms.set(Rooms.All, allUsers)
-        this.activeUsers.add(client.id)
-        client.broadcast.emit('confirmNewUser', ([userName, client.id, avatar, false]))
-        client.emit('confirmNewUser', ([userName, client.id, avatar, true]))
+    async newUser(client: Socket, [userName, avatar]: Array<string>) {
+        const user = await this.saveUser(client.id, avatar, userName)
+
+        client.broadcast.emit('confirmNewUser', ([user.userName, user.socketId, user.avatar, false]))
+        client.emit('confirmNewUser', ([user.userName, user.socketId, user.avatar, true]))
     }
 
     @SubscribeMessage('newRoom')
-    newRoom(client: Socket, roomName: string): WsResponse {
-        if (this.rooms.has(roomName)) {
+    async newRoom(client: Socket, roomName: string): Promise<WsResponse> {
+        const rooms = await this.appsService.getRoomNames()
+        const roomNames = rooms.map(({ roomName }) => roomName)
+
+        if (roomNames.includes(roomName)) {
             return null
         }
 
-        const [currentUserName] = this.users.get(client.id)
-
-        this.rooms.set(roomName, [currentUserName])
-        this.messages.set(roomName, [])
+        await this.appsService.createRoom(roomName, false)
```

Comment:
Promise.all ?

Reply:
to work properly appsService.addUserToRoom() requires completed appsService.createRoom() , thus Promise.all in this case might be invalid

Reply:
:+1: